server/dfr3-service/src/main/java/edu/illinois/ncsa/incore/service/dfr3/controllers/FragilityController.java

```diff
@@ -162,6 +169,36 @@ public FragilityController(
     public FragilitySet uploadFragilitySet(@ApiParam(value = "json representing the fragility set") FragilitySet fragilitySet) {
 
         UserInfoUtils.throwExceptionIfIdPresent(fragilitySet.getId());
+
+        // check if demand type is correct according to the definition; for now get the first definition
+        JSONObject demandDefinition = new JSONObject(commonRepository.getAllDemandDefinitions().get(0).toJson());
+        String hazardType = fragilitySet.getHazardType();
+        List<String> demandTypes = fragilitySet.getDemandTypes();
+        List<String> demandUnits = fragilitySet.getDemandUnits();
+
+        // check if size of demand type matches the unit
+        if (demandTypes.size() != demandUnits.size()) {
+            throw new IncoreHTTPException(Response.Status.BAD_REQUEST, "Demand Types should match the shape of Demand Units!");
+        } else {
+            Iterator<String> dt = demandTypes.iterator();
```

Comment:
Not for this PR, but I wonder if some of this code could be moved to a common utility because I can see the hazard service needing to do the same thing with a list of demand types and demand units and iterating over it to see if it's valid or not. The challenge might be the exceptions being thrown are trying to help the user understand what went wrong and that can be service specific (e.g. demand type is wrong for the fragility you are trying to create versus the demand request is wrong for the hazard).